cmd/bootstrap/group_replication_mysqlsh.go

```diff
@@ -0,0 +1,296 @@
+	if err != nil {
+		return errors.Wrapf(err, "add instance stdout: %s stderr: %s", stdout.String(), stderr.String())
+	}
+
+	return nil
+}
+func (m *mysqlsh) rejoinInstance(ctx context.Context, instanceDef string) error {
+	stdout, stderr, err := m.run(ctx, fmt.Sprintf("dba.getCluster('%s').rejoinInstance('%s')", m.clusterName, instanceDef))
+	if err != nil {
+		return errors.Wrapf(err, "rejoin instance stdout: %s stderr: %s", stdout.String(), stderr.String())
+	}
+
+	return nil
+}
+
+func (m *mysqlsh) rescanCluster(ctx context.Context) error {
+	stdout, stderr, err := m.run(ctx, fmt.Sprintf("dba.getCluster('%s').rescan({'addInstances': 'auto', 'removeInstances': 'auto'})", m.clusterName))
+	if err != nil {
+		return errors.Wrapf(err, "rescan cluster stdout: %s stderr: %s", stdout.String(), stderr.String())
+	}
+
+	return nil
+}
+
+func connectToLocal(ctx context.Context) (*mysqlsh, error) {
+	fqdn, err := getFQDN(os.Getenv("SERVICE_NAME"))
+	if err != nil {
+		return nil, errors.Wrap(err, "get FQDN")
+	}
+
+	return newShell(fqdn), nil
+}
+
+func connectToCluster(ctx context.Context, peers sets.Set[string]) (*mysqlsh, error) {
+	for _, peer := range peers.UnsortedList() {
+		shell := newShell(peer)
+		stdout, stderr, err := shell.run(ctx, fmt.Sprintf("dba.getCluster('%s')", shell.clusterName))
+		if err != nil {
+			log.Printf("Failed get cluster from peer %s, stdout: %s stderr: %s", peer, stdout.String(), stderr.String())
+			continue
+		}
+
+		return shell, nil
+	}
+
+	return nil, errors.New("failed to open connection to cluster")
+}
+
+func bootstrapGroupReplicationMySQLShell(ctx context.Context) error {
+	timer := stopwatch.NewNamedStopwatch()
+	err := timer.Add("total")
+	if err != nil {
+		return errors.Wrap(err, "add timer")
+	}
+	timer.Start("total")
+
+	defer func() {
+		timer.Stop("total")
+		log.Printf("bootstrap finished in %f seconds", timer.ElapsedSeconds("total"))
+	}()
+
+	exists, err := lockExists()
+	if err != nil {
+		return errors.Wrap(err, "lock file check")
+	}
+	if exists {
+		log.Printf("Waiting for bootstrap.lock to be deleted")
+		if err = waitLockRemoval(); err != nil {
+			return errors.Wrap(err, "wait lock removal")
+		}
+	}
+
+	log.Println("Bootstrap starting...")
+
+	localShell, err := connectToLocal(ctx)
+	if err != nil {
+		return errors.Wrap(err, "connect to local")
+	}
+
+	err = localShell.configureLocalInstance(ctx)
+	if err != nil {
+		return err
+	}
+	log.Printf("Instance (%s) configured to join to the InnoDB cluster", localShell.host)
+
+	podName, err := os.Hostname()
+	if err != nil {
+		return errors.Wrap(err, "get pod hostname")
+	}
+
+	peers, err := lookup(os.Getenv("SERVICE_NAME"))
+	if err != nil {
+		return errors.Wrap(err, "lookup")
+	}
+	log.Printf("peers: %v", peers.UnsortedList())
+
+	shell, err := connectToCluster(ctx, peers)
+	if err != nil {
+		if peers.Len() == 1 {
```

Comment:
Can you perform this check and create a cluster before calling `connectToCluster`?

Reply:
this checks depends on the fact that connecting to cluster is failed. we can have a single peer and be able to connect to the cluster.